lib/transaction/transaction.js

```diff
@@ -205,6 +205,10 @@ Transaction.prototype._isInvalidSatoshis = function() {
 };
 
 Transaction.prototype._hasFeeError = function(opts) {
+  if (this._getUnspentValue() < 0) {
```

Comment:
We could, but that doesn't answer my question. Is the cost of _getUnspentValue() what you were worried about or was it something else? Were you suggesting to merge _hasFeeError() and _hasMoreOutputThanInput() into one method?
I think that would make things more complicated rather than simpler. I just did the previous pull request in order to split all of that to reduce the cyclomatic complexity that lint was complaining about. I'd rather not undo that work.

If you are worried about calling _getUnspentValue() multiple times, I'd rather cache the result than merging several small methods into one big method. The fee checks also call _getUnspentValue() separately so it's not just here and in _hasMoreOutputThanInput().